src/librustc/ty/print/pretty.rs

```diff
@@ -943,7 +986,16 @@ pub trait PrettyPrinter<'tcx>:
             },
             (ConstValue::Scalar(Scalar::Raw { data, .. }), ty::Char) =>
                 p!(write("{:?}", ::std::char::from_u32(data as u32).unwrap())),
-            (ConstValue::Scalar(_), ty::RawPtr(_)) => p!(write("{{pointer}}")),
+            (ConstValue::Scalar(value), ty::Ref(..)) |
+            (ConstValue::Scalar(value), ty::RawPtr(_)) => {
+                match value {
```

Comment:
So this is a "fallback" case now, right? We already have a big case for references above. Unfortunately it is not immediately clear what distinguishes these cases.

Could you add some comments both at the `if let ty::Ref` above and at the match arm here describing why we do the matches we do in the order they are, and which various cases of references each conditional covers?

Reply:
I don't know the history of that big `if let ty::Ref` statement or why it looks the way it does internally, so I don't feel comfortable documenting it. But I added a comment specifying that it needs to happen before the general case.

Reply:
From what I can see, the point of that is to print the *content* of the slice/array when possible for types `&[u8]`, `[u8; N]` and `&str`. Do you agree? If yes, please write that. Right now you only mention `&str` but that is not the only type handled here.

The fallback then is that for all other references and raw pointers, we just print the pointer *value*, not the thing it points to.